### PR TITLE
Add pre-commit running bandit, codespell, flake8

### PR DESCRIPTION
**Pull request by cclauss**
Add https://pre-commit.com and a GitHub Action to run these tests on all pull requests.

Contributors can run these tests locally on every commit by doing:
`python3 -m pip install pre-commit ; pre-commit install`

Once this lands, we can remove bandit and codespell from the lint_python GitHub Action.

### CODE REVIEW

**Comment by MichaIng** [comment]
I was just wondering what the benefit of `pre-commit` is. So it is that one can run the same CI/CD checks locally?

Btw, I noticed the issue with using push + pull_request triggers that we often run two instances of the workflow concurrently, one for the push and one for the opened PR. On my main project repos we trigger only on push to avoid this, but it has the downside that PRs from forks are not tested on our repo but only on the forks repo, so it is more difficult for us to review results. Not sure whether there is a clever way to trigger on push only for origin branches and on pull_request only for foreign branches?

**Comment by cclauss** [comment]
Using pre-commit in this way ensures identical results when run locally and in CI.
It also ensures that if a contributor fails to install pre-commit locally, our CI will highlight any issues in their submissions.

**Comment by cclauss** [comment]
If you can figure out https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onevent_nametypes then you are way ahead of me.

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# https://pre-commit.com
+# This GitHub Action assumes that the repo contains a valid .pre-commit-config.yaml file.
+name: pre-commit
+on:
+  pull_request:
+  push:
+    branches: [master]
```
Comment (MichaIng) [inline .github/workflows/pre-commit.yml:7]: I think it should trigger on all branches, so when someone is doing a push to a feature branch, or we merge into the dev branch, CI/CD is always triggered to get new results?

**Comment by MichaIng** [comment]
Now we also see the result on your fork: https://github.com/cclauss/motioneye/actions/workflows/pre-commit.yml

> fix end of files.....................................(no files to check)Skipped

No files to check? 🤔

**Comment by MichaIng** [comment]
Okay, understood. So no files to check because all file endings are fine already.

**Comment by cclauss** [comment]
I don't have Merge rights so if you approve, please merge.

**Pull request merged**